lnwallet/chancloser/chancloser_test.go

```diff
@@ -535,66 +528,81 @@ func TestTaprootFastClose(t *testing.T) {
+	oClosingSigned, err := bobCloser.BeginNegotiation()
+	require.NoError(t, err)
+	tx, _ := bobCloser.ClosingTx()
+	require.Nil(t, tx)
+	require.True(t, oShutdown.IsSome())
+	require.True(t, oClosingSigned.IsNone())
+
+	bobShutdown := oShutdown.UnsafeFromSome()
 
 	// Alice should process the shutdown message, and create a closing
 	// signed of her own.
-	aliceMsgs, closeFinished, err := aliceCloser.ProcessCloseMsg(bobMsgs[0])
+	oShutdown, err = aliceCloser.ReceiveShutdown(bobShutdown)
 	require.NoError(t, err)
-	require.False(t, closeFinished)
-	require.Len(t, aliceMsgs, 1)
-	require.IsType(t, &lnwire.ClosingSigned{}, aliceMsgs[0])
+	oClosingSigned, err = aliceCloser.BeginNegotiation()
+	require.NoError(t, err)
+	tx, _ = aliceCloser.ClosingTx()
+	require.Nil(t, tx)
+	require.True(t, oShutdown.IsNone())
+	require.True(t, oClosingSigned.IsSome())
+
+	aliceClosingSigned := oClosingSigned.UnsafeFromSome()
 
 	// Next, Bob will process the closing signed message, and send back a
 	// new one that should match exactly the offer Alice sent.
-	bobMsgs, closeFinished, err = bobCloser.ProcessCloseMsg(aliceMsgs[0])
+	oClosingSigned, err = bobCloser.ReceiveClosingSigned(aliceClosingSigned)
 	require.NoError(t, err)
-	require.True(t, closeFinished)
-	require.Len(t, aliceMsgs, 1)
-	require.IsType(t, &lnwire.ClosingSigned{}, bobMsgs[0])
+	tx, _ = bobCloser.ClosingTx()
+	require.NotNil(t, tx)
+	require.True(t, oClosingSigned.IsSome())
+
+	bobClosingSigned := oClosingSigned.UnsafeFromSome()
 
 	// At this point, Bob has accepted the offer, so he can broadcast the
 	// closing transaction, and considers the channel closed.
 	_, err = lnutils.RecvOrTimeout(broadcastSignal, time.Second*1)
 	require.NoError(t, err)
 
 	// Bob's fee proposal should exactly match Alice's initial fee.
-	aliceOffer := assertType[*lnwire.ClosingSigned](t, aliceMsgs[0])
-	bobOffer := assertType[*lnwire.ClosingSigned](t, bobMsgs[0])
-	require.Equal(t, aliceOffer.FeeSatoshis, bobOffer.FeeSatoshis)
+	require.Equal(
+		t, aliceClosingSigned.FeeSatoshis, bobClosingSigned.FeeSatoshis,
+	)
 
 	// If we modify Bob's offer, and try to have Alice process it, then she
 	// should reject it.
-	ogOffer := bobOffer.FeeSatoshis
-	bobOffer.FeeSatoshis /= 2
+	ogOffer := bobClosingSigned.FeeSatoshis
+	bobClosingSigned.FeeSatoshis /= 2
 
-	_, _, err = aliceCloser.ProcessCloseMsg(bobOffer)
+	_, err = aliceCloser.ReceiveClosingSigned(bobClosingSigned)
 	require.Error(t, err)
 	require.Contains(t, err.Error(), "was not accepted")
 
 	// We'll now restore the original offer before passing it on to Alice.
-	bobOffer.FeeSatoshis = ogOffer
+	bobClosingSigned.FeeSatoshis = ogOffer
 
 	// If we use the original offer, then Alice should accept this message,
 	// and finalize the shutdown process. We expect a message here as Alice
 	// will echo back the final message.
-	aliceMsgs, closeFinished, err = aliceCloser.ProcessCloseMsg(bobMsgs[0])
+	oClosingSigned, err = aliceCloser.ReceiveClosingSigned(bobClosingSigned)
```

Comment:
Other info leakage here now is that the tests are made _slightly_ more white box. Could imagine that if `ProcessCloseMsg` is kept _mostly_ in place, then the new co-op close state machine can re-use the overall tests flow, even if the message dispatch is slightly diff. 

Ultimately, I think we're all heading closer to the idea of extracting the routing/dispatch logic from the peer, into something else with an additional layer of indirection. Right now the peer ends up knowing a lot about the channel life cycle (new channels, pending channels, etc, etc), when really all it needs to do is read off the wire, and figure out who/what to pass the message to then go back to the top.

Feels like we're sort of reaching diminishing returns with this part of the review conversation, and in the near-ish future, we can erect the more abstract message router/dispatcher which will once again hide more of the information from the peer.